pkg/replicaset/replicaset.go

```diff
@@ -0,0 +1,452 @@
+
+	if len(specs) == 0 {
+		glog.V(100).Infof("The container specs are empty")
+
+		builder.errorMsg = "cannot accept empty list as container specs"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.Template.Spec.Containers == nil {
+		builder.Definition.Spec.Template.Spec.Containers = specs
+	} else {
+		builder.Definition.Spec.Template.Spec.Containers = append(builder.Definition.Spec.Template.Spec.Containers, specs...)
+	}
+
+	return builder
+}
+
+// Exists checks whether the given replicaset exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if replicaset %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get fetches the defined replicaset from the cluster.
+func (builder *Builder) Get() (*appsv1.ReplicaSet, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting replicaset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	replicaSetObj := &appsv1.ReplicaSet{}
+	err := builder.apiClient.Get(context.TODO(),
+		goclient.ObjectKey{
+			Name:      builder.Definition.Name,
+			Namespace: builder.Definition.Namespace,
+		}, replicaSetObj)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return replicaSetObj, nil
+}
+
+// Create builds replicaset in the cluster and stores the created object in struct.
+func (builder *Builder) Create() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating replicaset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Update renovates the existing replicaset object with replicaset definition in builder.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating replicaset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	if err == nil {
+		builder.Object = builder.Definition
+	}
+
+	return builder, err
+}
+
+// Delete removes the replicaset.
+func (builder *Builder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting replicaset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return nil
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Object)
+
+	if err != nil {
+		return err
+	}
+
+	builder.Object = nil
+
+	return err
```

Comment:
same question

Reply:
In this case it could be nil because we return real error on line 326 but it you want you can leave it as is. I agree in terms of visibility nil is better however functionality wise is the same

Reply:
At this point the `err != nil` has been filtered so explicitly returning `nil` is okay.